File: DartLibraryGenerator/model/local_model.dart
import 'package:collection/collection.dart';

import 'local_model_entity.dart';

class LocalModel {
  List<LocalModelEntity> entities = [];

  LocalModelEntity operator[](int id) {
    return 
      entities.firstWhereOrNull((el) => el.declaration.id == id)  ?? 
      (() => throw new UnsupportedError("Entity with id '$id' is not found in the local model."))();
  }
}

File: NetLibraryGenerator/NetLibraryGenerator/Model/LocalModel.cs
namespace NetLibraryGenerator.Model
{
    public class LocalModel
    {
        public LocalModelEntity this[int index]
        {
            get {
                return Responses.FirstOrDefault(r => r.Declaration.Id == index) ??
                       Parameters.FirstOrDefault(p => p.Declaration.Id == index) ??
                       throw new KeyNotFoundException($"Entity with id {index} not found in local model.");
            }
        }

        public List<LocalModelEntity> Parameters { get; } = new();
        public List<LocalModelEntity> Responses { get; } = new();
    }
}
